go/extractor/cli/go-autobuilder/go-autobuilder.go

```diff
@@ -692,3 +695,161 @@ func main() {
 
 	extract(depMode, modMode)
 }
+
+const minGoVersion = "1.11"
+const maxGoVersion = "1.20"
+
+func outsideSupportedRange(version string) bool {
+	short := semver.MajorMinor("v" + version)
+	return semver.Compare(short, "v"+minGoVersion) < 0 || semver.Compare(short, "v"+maxGoVersion) > 0
+}
+
+func checkForUnsupportedVersions(v versionInfo) (msg, version string) {
+	if v.goDirectiveFound && outsideSupportedRange(v.goModVersion) {
+		msg = "The version of Go found in the `go.mod` file (" + v.goModVersion +
+			") is outside of the supported range (" + minGoVersion + "-" + maxGoVersion + ")."
+		version = ""
```

Comment:
Would it be better to return `nil` instead of an empty string if these functions cannot determine which go version to install? And have `msg` as an `error` rather than a `string`?

Reply:
I like the `nil` idea but strings can't be `nil` in go. I have effectively given the empty string a special meaning, and I've been careful to always assign a value to `version` even when I could leave it as the default value so that it's easier to spot any places where it hasn't been considered.

Reply:
I don't think `msg` should be an `error` because then people will assume that it's only non-`nil` if something has gone wrong, whereas it's actually non-`nil` on every path through the code.